app/src/main/java/org/akvo/flow/presentation/navigation/ViewSurvey.java

```diff
@@ -51,4 +71,17 @@ public boolean isMonitored() {
     public String getRegistrationSurveyId() {
         return registrationSurveyId;
     }
+
+    @Override
+    public int describeContents() {
+        return 0;
+    }
+
+    @Override
+    public void writeToParcel(Parcel dest, int flags) {
+        dest.writeLong(id);
+        dest.writeString(name);
+        dest.writeByte((byte) (isMonitored ? 1 : 0));
+        dest.writeString(registrationSurveyId);
```

Comment:
The registrationSurveyId is currently set as a string? Do we want to change this to a long?